### PR TITLE
[ACR] Type renames from Arch Board review

### LINKED ISSUE #34958: [ACR] Follow-ups from Arch Board review
This issue tracks the work for .NET coming out of the [ACR arch board review](https://github.com/Azure/azure-sdk/issues/5643).

Per comments from the [.NET APIView](https://apiview.dev/Assemblies/Review/7ebfcb36ff6149ba96170bd059094513/f989159861d846b3abb4384d01a27da9?diffRevisionId=500d32ab2730498084dedadf61d740fc&doc=False&diffOnly=True):
- [x] Move client out of the `.Specialized` namespace
- [ ] Rename client
- [x] Change `RepositoryName` property to `Repository` (or be consistent with the name of the parameter passed to the constructor)
- [x] Update `DownloadBlobTo()` doc comments to clarify return type (use Storage as an example)
- [x] Update doc comment on `UploadBlob` to clarify that RawResponse returned is the last of the chunks
- [x] Rename parameters to `content` in `UploadBlob` methods
- [x] Rename parameters to `manifest` in `UploadManifest` methods
- [x] Remove `UploadManifest` overload that takes a Stream
- [ ] Rename `DownloadBlobResult` to something Registry-specific
- [ ] Rework how deserialization to a strong type for manifest works
- [x] Change type of `OciAnnotations.Url` to `Uri`
- [ ] Consider adding an extensible enum of known media types for `OciDescriptor.MediaType`
- [ ] Move `schemaVersion=2` into a swagger constant
- [x] Change `OciImageManifest.Config` to `Configuration`

### PR DESCRIPTION
**Pull request by annelo-msft**
This PR implements the following changes:
- Renames `ContainerRegistryBlobClient` to `ContainerRegistryContentClient`
- Renames `DownloadBlobResult` to `DownloadRegistryBlobResult`
- Renames `DownloadBlobStreamingResult` to `DownloadRegistryBlobStreamingResult`
- Renames `UplaodBlobResult` to `UploadRegistryBlobResult`
- Removes the default value for `OciImageManifest.SchemaVersion`
- Renames `DownloadManifest` to `GetManifest`
- Renames `UploadManifest` to `SetManifest`
- Changes `GetManifestResult.Content` to `GetManifestResult.Manifest`
- Removes `GetManifestResult.AsOciManifest()` in favor of `BinaryData.ToObjectFromJson<OciImageManifest>()`

Fixes https://github.com/Azure/azure-sdk-for-net/issues/34958

### CODE REVIEW

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/api/Azure.Containers.ContainerRegistry.netstandard2.0.cs [hunk 1]
```diff
@@ -231,26 +235,25 @@ public partial class ContainerRepositoryProperties
         public string RegistryLoginServer { get { throw null; } }
         public int TagCount { get { throw null; } }
     }
-    public partial class DownloadBlobResult
-    {
-        internal DownloadBlobResult() { }
-        public System.BinaryData Content { get { throw null; } }
-        public string Digest { get { throw null; } }
-    }
     public partial class DownloadBlobStreamingResult : System.IDisposable
```
Comment (annelo-msft) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/api/Azure.Containers.ContainerRegistry.netstandard2.0.cs:238]: This should change to DownloadRegistryBlobStreamingResult

Comment (annelo-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/api/Azure.Containers.ContainerRegistry.netstandard2.0.cs]: Complete.

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/samples/Sample05_UploadDownloadImage.md [hunk 2]
```diff
@@ -161,17 +161,17 @@ else if (result.MediaType == "application/vnd.oci.image.index.v1+json")
 A manifest can be deleted as shown below.  It is also possible to delete a full image using the `ContainerRegistryClient` as shown in [Sample 2: Delete Image](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/containerregistry/Azure.Containers.ContainerRegistry/samples/Sample02b_DeleteImagesAsync.md).
 
 ```C# Snippet:ContainerRegistry_Samples_DeleteManifest
-DownloadManifestResult downloadManifestResult = await client.DownloadManifestAsync(tag);
-await client.DeleteManifestAsync(downloadManifestResult.Digest);
+GetManifestResult GetManifestResult = await client.GetManifestAsync(tag);
+await client.DeleteManifestAsync(GetManifestResult.Digest);
```
Comment (lmolkova) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/samples/Sample05_UploadDownloadImage.md:165]:
```suggestion
GetManifestResult getManifestResult = await client.GetManifestAsync(tag);
await client.DeleteManifestAsync(getManifestResult.Digest);
```

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Content/GetManifestResult.cs [hunk 3]
```diff
@@ -31,17 +29,8 @@ internal DownloadManifestResult(string digest, ManifestMediaType mediaType, Bina
         public ManifestMediaType MediaType { get; }
 
         /// <summary>
-        /// The serialized content that was downloaded.
+        /// The serialized manifest.
```
Comment (lmolkova) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Content/GetManifestResult.cs:32]: it's not serialized, is it?

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Content/ContainerRegistryContentClient.cs [hunk 4]
```diff
@@ -533,32 +533,32 @@ private static bool TryGetLength(Stream content, out long length)
...
+        /// Gets a manifest.
         /// </summary>
-        /// <param name="tagOrDigest">The tag or digest of the manifest to download.</param>
-        /// <param name="mediaTypes">The set of media types to accept for the manifest being downloaded.</param>
+        /// <param name="tagOrDigest">The tag or digest of the manifest to get.</param>
+        /// <param name="mediaTypes">The set of media types to accept for the manifest.</param>
         /// <param name="cancellationToken">The cancellation token to use.</param>
         /// <returns>The download manifest result.</returns>
-        public virtual Response<DownloadManifestResult> DownloadManifest(string tagOrDigest, IEnumerable<ManifestMediaType> mediaTypes, CancellationToken cancellationToken = default)
+        public virtual Response<GetManifestResult> GetManifest(string tagOrDigest, IEnumerable<ManifestMediaType> mediaTypes, CancellationToken cancellationToken = default)
```
Comment (lmolkova) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Content/ContainerRegistryContentClient.cs:542]: out-of-scope of this Pr, but based on the feedback from the service team, we should remove this and do now allow requesting specific types

**Review by lmolkova — APPROVED** [review]

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Content/GetManifestResult.cs (hunk 3 above)
Comment (annelo-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Content/GetManifestResult.cs:32]: The BinaryData property holds the raw JSON and not a .NET model type.  Do you not call this serialized?

Comment (annelo-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Content/GetManifestResult.cs:32]: Happy to change, though - I'll go look at what Java has and copy.

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Content/ContainerRegistryContentClient.cs (hunk 4 above)
Comment (annelo-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Content/ContainerRegistryContentClient.cs:542]: Filed https://github.com/Azure/azure-sdk-for-net/issues/35083 to track.

**Pull request merged**